File: 2021/01/17/jh20s/contest#222/1710.cpp
class Solution {
public:
int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {

	int ans = 0;
	vector<pair<int, int>> v;

	for (int i = 0; i < boxTypes.size(); i++) {
		v.push_back({ boxTypes[i][1], boxTypes[i][0] });
	}
	sort(v.begin(), v.end(),greater<pair<int,int>>());

	for (int i = 0; i < v.size() && truckSize>0; i++) {
		int cnt = min(v[i].second, truckSize);
		ans += cnt * v[i].first;
		truckSize -= cnt;
    }
	return ans;
}
};

File: 2021/01/17/jh20s/contest#222/1710.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        res = 0
        for box in sorted(boxTypes, key=lambda x: x[1], reverse = True):
            res += min(box[0], truckSize) * box[1]
            truckSize = truckSize - box[0]
            if truckSize <=0:
                break
        
        return res